kernel/src/engine/arrow_expression.rs

```diff
@@ -318,27 +363,21 @@ fn evaluate_expression(
                 .map(wrap_comparison_result)
                 .map_err(Error::generic_err)
         }
+        (BinaryOperation { op, left, right }, _)
+            if matches!(**left, MapAccess { .. }) && matches!(**right, Literal(_)) =>
```

Comment:
Why is it important that we handle BinaryOperation with left: MapAccess, and right: Literal? What about the other way around with left: Literal, right: MapAccess?

Reply:
Conceptually, we're doing `map['key']`; `key[map]` doesn't really make sense?

Reply:
Well the left side is `map['key']` by itself, so if you knew the key you'd write `map['key'] = 1` or whatever so the map access left side essentially boils down to a literal vs literal match.